src/rulesets/Base/arraymath.jl

```diff
@@ -40,6 +40,27 @@ function rrule(
     return A * B, times_pullback
 end
 
+function rrule(
+    ::typeof(*),
+    A::AbstractVector{<:CommutativeMulNumber},
+    B::AbstractMatrix{<:CommutativeMulNumber},
+)
+    function times_pullback(Ȳ)
```

Comment:
```suggestion
    function times_pullback(Ȳ)
        @assert size(B, 1) === 1   # otherwise primal would have failed.
```

Reply:
(I think `==` would be better here, because `AbstractArray`s might not necessarily use `Int` for indexing.)